internal/ratelimit/map.go

```diff
@@ -0,0 +1,64 @@
+package ratelimit
+
+import (
+	"net/http"
+	"time"
+)
+
+// Map maps categories to rate limit deadlines.
+//
+// A rate limit is in effect for a given category if either the category's
+// deadline or the deadline for the special CategoryAll has not yet expired.
+//
+// Use IsRateLimited to check whether a category is rate-limited.
+type Map map[Category]Deadline
```

Comment:
Yeah so originally I called this `RateLimits`.

But when you look at a call site it reads `ratelimit.RateLimits`, which [stutters](https://blog.golang.org/package-names#TOC_3.).
Since the documentation talked about "mapping", I renamed to `ratelimit.Map`.

Note that despite being exported, it is in an `internal/` package, so it can only be used by the SDK itself, and it not anywhere else. It is not public API.

I wanted to reserve the right to change this completely without API changes. Using an `internal/` package allows us to organize the code in such a way that all rate limit-related code is grouped together in one place instead of mixed up with the other existing files in the top-level package.

In particular, it could have been implemented using a different data structure than a `map` -- note that since we only want to store a fixed number of categories, a `struct` would do (but didn't seem worth pursuing that change right now: first make it work, later make it faster/less memory if we need to).